src/Compilers/CSharp/Portable/Symbols/Source/SimpleProgramNamedTypeSymbol.cs

```diff
@@ -224,7 +224,7 @@ protected override MembersAndInitializers BuildMembersAndInitializers(Diagnostic
                                               instanceInitializersSyntaxLength: 0);
         }
 
-        public override bool IsImplicitlyDeclared => true;
+        public override bool IsImplicitlyDeclared => false;
```

Comment:
>false [](start = 53, length = 5)

It seems surprising that this type is not marked implicitly declared. (Note, the comment on the `class` indicates that it is implicitly declared.)

Reply:
> It seems surprising that this type is not marked implicitly declared. (Note, the comment on the class indicates that it is implicitly declared.)

It seems that IDE features and AnalyzerDriver have strong dependency on the fact that anything containing user-written code returns true from this API. So, yes the type doesn't have explicit TypeDeclarationSyntax, but to follow the established dependencies, we return false because it contains user-written code.

---
In reply to: [433040558](https://github.com/dotnet/roslyn/pull/44716#discussion_r433040558) [](ancestors = 433040558)

Reply:
It does feel odd that this is required, considering that for example synthesized constructors are considered to be "implicitly declared" but they contain user-written field initializers.

Reply:
> It does feel odd that this is required, considering that for example synthesized constructors are considered to be "implicitly declared" but they contain user-written field initializers.

Synthesized constructors do not contain user-written code, field initializers are not "inside" the constructor, they are associated with fields rather that with the constructor

---
In reply to: [433351758](https://github.com/dotnet/roslyn/pull/44716#discussion_r433351758) [](ancestors = 433351758)